src/test/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacadeIT.java

```diff
@@ -432,4 +437,35 @@ public void assignBulkEndpoint_scheduleSingleAssignmentWithScheduledDateAfterDue
         // check status code is NO_CONTENT (successful)
         assertEquals(Response.Status.NO_CONTENT.getStatusCode(), deleteAssignmentResponse.getStatus());
     }
+
+    @Test public void getAssignmentProgressDownloadCSV_succeeds() throws Exception {
+        // log in as Test teacher, create request
+        LoginResult teacherLogin = loginAs(httpSession, ITConstants.TEST_TEACHER_EMAIL,
+                ITConstants.TEST_TEACHER_PASSWORD);
+        HttpServletRequest downloadAssignmentRequest = createRequestWithCookies(new Cookie[]{teacherLogin.cookie});
+        replay(downloadAssignmentRequest);
+
+        Response downloadAssignmentResponse = assignmentFacade.getAssignmentProgressDownloadCSV(downloadAssignmentRequest, 4L, "excel");
+        String downloadAssignmentContents = downloadAssignmentResponse.getEntity().toString();
+        
+        // Note that the first two lines have a trailing whitespace. If your IDE automatically removes these from the expected file it may cause a mismatch
```

Comment:
This normally only happens during creation or modification of a line, unless you've configured your IDE to behave differently. Now that the file has been created, your IDE shouldn't remove files that you haven't changed, so it's safe to remove this comment.